docs/misc/math-expr.md

```diff
@@ -63,7 +63,7 @@ The following built-in functions are available.
 
 |name|description|
 |----|-----------|
-|cast|cast(expr,'LONG' or 'DOUBLE' or 'STRING' or 'LONG_ARRAY', or 'DOUBLE_ARRAY' or 'STRING_ARRAY') returns expr with specified type. exception can be thrown. Scalar types may be cast to array types and will take the form of a single element list (null will still be null). |
+|cast|cast(expr,'LONG' or 'DOUBLE' or 'STRING' or 'ARRAY<LONG>', or 'ARRAY<DOUBLE>' or 'ARRAY<STRING>') returns expr with specified type. exception can be thrown. Scalar types may be cast to array types and will take the form of a single element list (null will still be null). |
```

Comment:
Is this a backwards compatibility issue (did something really change) or fixing a doc bug?

Reply:
It was documenting the old type syntax, which still works but shouldn't be encouraged I think, so i updated it

Reply:
well, i did it in the wrong PR, but added some tests to make sure i knew what i was talking about https://github.com/apache/druid/pull/12971/files#diff-8f872e230b0833df3a50ab4c9ef8283346e78fff991e0c3e53beb693329679d7R371